actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -822,7 +822,7 @@ def url_for(options, route_name = nil, url_strategy = UNKNOWN, method_name = nil
         path = route_with_params.path(method_name)
         params = route_with_params.params
 
-        if options.key? :params
+        if options.key?(:params) && options[:params].is_a?(Hash)
```

Comment:
Shouldn't we check the second branch first?
```suggestion
        if options[:params].is_a?(Hash) && options.key?(:params)
```

Reply:
if `options[:params].is_a?(Hash)` evaluates to true, you know that `options.key?(:params)` will also be true. So the line _could_ be reduced to just `if options[:params].is_a?(Hash)`.

Reply:
Whoops, I didn't notice the check was `options[:params].is_a?(Hash)` not `options.is_a?(Hash)`.

Reply:
@bquorning your suggestion is on point 👌🏼 I changed accordingly